dotnet/samples/ApplicationInsightsExample/Program.cs

```diff
@@ -33,75 +30,61 @@ public sealed class Program
     /// </remarks>
     private const LogLevel MinLogLevel = LogLevel.Information;
 
+    /// <summary>
+    /// Instance of <see cref="ActivitySource"/> for the application activities.
+    /// </summary>
+    private static readonly ActivitySource s_activitySource = new("ApplicationInsights.Example");
+
     /// <summary>
     /// The main entry point for the application.
     /// </summary>
     /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
     public static async Task Main()
     {
-        var serviceProvider = GetServiceProvider();
+        var connectionString = Env.Var("ApplicationInsights__ConnectionString");
 
-        var telemetryClient = serviceProvider.GetRequiredService<TelemetryClient>();
-        var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
+        using var traceProvider = Sdk.CreateTracerProviderBuilder()
+            .AddSource("Microsoft.SemanticKernel.*")
+            .AddSource("ApplicationInsights.Example")
+            .AddAzureMonitorTraceExporter(options => options.ConnectionString = connectionString)
+            .Build();
 
-        using var meterListener = new MeterListener();
-        using var activityListener = new ActivityListener();
+        using var meterProvider = Sdk.CreateMeterProviderBuilder()
+            .AddMeter("Microsoft.SemanticKernel.*")
+            .AddAzureMonitorMetricExporter(options => options.ConnectionString = connectionString)
+            .Build();
 
-        ConfigureMetering(meterListener, telemetryClient);
-        ConfigureTracing(activityListener, telemetryClient);
+        using var loggerFactory = LoggerFactory.Create(builder =>
+        {
+            // Add OpenTelemetry as a logging provider
+            builder.AddOpenTelemetry(options =>
+            {
+                options.AddAzureMonitorLogExporter(options => options.ConnectionString = connectionString);
+                // Format log messages. This is default to false.
+                options.IncludeFormattedMessage = true;
+            });
+            builder.SetMinimumLevel(MinLogLevel);
+        });
 
         var kernel = GetKernel(loggerFactory);
         var planner = GetSequentialPlanner(kernel, loggerFactory);
+        // var planner = GetActionPlanner(kernel, loggerFactory);
```

Comment:
Any reason this commented code stays here?

Reply:
This example does have the function implemented so that people can switch to a different planner.